backend/app/routers/authorization.py

```diff
@@ -107,61 +110,68 @@ async def get_dataset_role(
         return auth_db.dict()
 
 
-@router.get("/datasets/{dataset_id}/role/viewer")
+@router.get("/datasets/{dataset_id}/role/viewer}")
 async def get_dataset_role_viewer(
-    dataset_id: str, allow: bool = Depends(Authorization("viewer"))
+        dataset_id: str,
+        admin_mode: bool = Depends(get_admin_mode),
+        allow: bool = Depends(Authorization("viewer")),
 ):
     """Used for testing only. Returns true if user has viewer permission on dataset, otherwise throws a 403 Forbidden HTTP exception.
     See `routers/authorization.py` for more info."""
     return {"dataset_id": dataset_id, "allow": allow}
 
 
-@router.get("/datasets/{dataset_id}/role/owner")
+@router.get("/datasets/{dataset_id}/role/owner}")
 async def get_dataset_role_owner(
-    dataset_id: str, allow: bool = Depends(Authorization("owner"))
+        dataset_id: str,
+        admin_mode: bool = Depends(get_admin_mode),
+        allow: bool = Depends(Authorization("owner")),
 ):
     """Used for testing only. Returns true if user has owner permission on dataset, otherwise throws a 403 Forbidden HTTP exception.
     See `routers/authorization.py` for more info."""
     return {"dataset_id": dataset_id, "allow": allow}
 
 
-@router.get("/files/{file_id}/role", response_model=RoleType)
+@router.get("/files/{file_id}/role}", response_model=RoleType)
 async def get_file_role(
-    file_id: str,
-    current_user=Depends(get_current_username),
-    role: RoleType = Depends(get_role_by_file),
-    admin=Depends(get_admin),
+        file_id: str,
+        admin_mode: bool = Depends(get_admin_mode),
+        current_user=Depends(get_current_username),
+        role: RoleType = Depends(get_role_by_file),
+        admin=Depends(get_admin),
 ):
-    # admin is a superuser and has all the privileges
-    if admin:
+    # admin is a superuser and has all the privileges, only show if the user has turned on the admin mode
+    if admin and admin_mode:
```

Comment:
This alone won't work because there is also role check on the dependencies `get_role_by_file`